free-2.md

```diff
@@ -1,44 +1,98 @@
+called "ImageOrganizer" (although I will probably change this name). It is
+designed as a simple DSL for organizing large quantities of images based on
+user defined categories. This DSL will help computer users, like myself,
+who generally save a lot of pictures to eventually need to attempt to sort
+them. Although in the past I've designed a simple GUI with buttons
+corresponding to predefined folders, I have always thought it would
+be interesting to build a simple command line language that can be interacted
+with by a customized GUI (similar to how Git GUIs interact with Git). Thus
+the command line tools will allow for simple workflow, classification,
+taging, macros, etc. Although many well defined tools already exist 
+for organizing pictures (even Facebook provides features like this), by
+building a simple scripting language like this, I can design different GUIs
+to be used locally on my computer or remotely on my Android cell phone.
+This means ideally you can organize your local files with a slick
+consistent underlying system with a GUI customized to the current
+enviorment.
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+This DSL is essentially an API that will allow for easy local or remote
+organization of locally stored pictures. This addresses the need by
+allowing users to organize local files regardless of their location
+(i.e. generally the best time to organize pictures, since it is a realitively
+mindless task, is when you're on a bus, on a line, etc. and not sitting at
+your computer where you could be doing something "more productive").
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+It is a project that would be very useful to me and one that I have made
+several solutions for in the past. Originally, I made the first GUI-only
+solution so my grandma, who is very uncomfortable with computers, could
+organize some digital pictures without too many excessive features or buttons
+that could be potentially confusing. After that I made a better GUI program
+with more features that I have used. Now I essentially want to make this
+DSL so I could run it as a server application with a remote GUIs.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Extendable language for image organization.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+Like Git, I imagine this program will be made up of a number of useful
+commands that are all launched from command line arguments. I imagine
+macros, the interesting DSL aspect of this project, will be defined
+in an init file similar to a .bashrc or .vimrc file. The syntax will
+allow for easy access to file system level actions like moving and copying
+files but also to accessing properties of pictures (i.e. date modified,
+average color, dimensions, etc.)
```

Comment:
I think this is a good strategy, but I also want to stress that (in my opinion) hiding as much of the file system level stuff from the user as possible is super important. I could see this project starting off neat but becoming too difficult to use for a normal user.